finality-aleph/src/network/gossip/service.rs

```diff
@@ -246,41 +270,60 @@ impl<N: RawNetwork, D: Data> Service<N, D> {
-        let peer_id = match self.random_peer(&peer_ids, protocol) {
+    fn send_to_random_authentication(&mut self, data: AD, peer_ids: HashSet<N::PeerId>) {
+        let peer_id = match self.random_peer(&peer_ids, Protocol::Authentication) {
             Some(peer_id) => peer_id.clone(),
             None => {
-                trace!(target: "aleph-network", "Failed to send to random peer, no peers are available.");
+                trace!(target: "aleph-network", "Failed to send authentication message to random peer, no peers are available.");
                 return;
             }
         };
-        self.send(data, peer_id, protocol);
+        self.send_authentication_data(data, peer_id);
     }
 
-    fn broadcast(&mut self, data: D, protocol: Protocol) {
-        let peers = self.protocol_peers(protocol).clone();
+    fn send_to_random_block_sync(&mut self, data: BSD, peer_ids: HashSet<N::PeerId>) {
+        let peer_id = match self.random_peer(&peer_ids, Protocol::BlockSync) {
+            Some(peer_id) => peer_id.clone(),
+            None => {
+                trace!(target: "aleph-network", "Failed to send block sync message to random peer, no peers are available.");
+                return;
+            }
+        };
+        self.send_block_sync_data(data, peer_id);
+    }
+
+    fn broadcast_authentication(&mut self, data: AD) {
+        let peers = self.protocol_peers(Protocol::Authentication).clone();
         for peer in peers {
-            self.send(data.clone(), peer, protocol);
+            self.send_authentication_data(data.clone(), peer);
         }
     }
 
-    fn handle_network_event(
-        &mut self,
-        event: Event<N::PeerId>,
-    ) -> Result<(), mpsc::TrySendError<(D, N::PeerId)>> {
+    fn broadcast_block_sync(&mut self, data: BSD) {
+        let peers = self.protocol_peers(Protocol::BlockSync).clone();
+        for peer in peers {
+            self.send_block_sync_data(data.clone(), peer);
+        }
+    }
+
+    fn handle_network_event(&mut self, event: Event<N::PeerId>) -> Result<(), ()> {
         use Event::*;
         match event {
             StreamOpened(peer, protocol) => {
                 trace!(target: "aleph-network", "StreamOpened event for peer {:?} and the protocol {:?}.", peer, protocol);
-                let rx = match &protocol {
+                match protocol {
                     Protocol::Authentication => {
                         let (tx, rx) = tracing_unbounded(
                             "mpsc_notification_stream_authentication",
                             QUEUE_SIZE_WARNING,
                         );
                         self.authentication_connected_peers.insert(peer.clone());
                         self.authentication_peer_senders.insert(peer.clone(), tx);
-                        rx
+                        self.spawn_handle.spawn(
+                            "aleph/network/peer_sender",
```

Comment:
Maybe they should use different names since they are used for different protocols? Can they?

Reply:
I'm pretty sure they can and good idea, added prefixes.